runner_manager/backend/aws.py

```diff
@@ -16,66 +18,130 @@ class AWSBackend(BaseBackend):
     instance_config: AWSInstanceConfig
 
     @property
-    def client(self) -> client:
-        """Returns a AWS Compute Engine client."""
+    def client(self) -> EC2Client:
+        """Return a AWS Compute Engine client."""
         return client("ec2", region_name=self.config.region)
 
     def create(self, runner: Runner) -> Runner:
         """Create a runner."""
         labels: Dict[str, str] = {}
         if self.manager:
             labels["runner-manager"] = self.manager
+        self.instance_config.subnet_id = self.config.subnet_id
         instance_resource: Dict = self.instance_config.configure_instance(runner)
         try:
             instance = self.client.run_instances(**instance_resource)
             runner.instance_id = instance["Instances"][0]["InstanceId"]
-        except ClientError as e:
+            # Wait for instance to be running
+            waiter = self.client.get_waiter("instance_running")
+            waiter.wait(InstanceIds=[runner.instance_id])
+        except Exception as e:
+            log.error(e)
             raise e
         return super().create(runner)
 
     def delete(self, runner: Runner):
         """Delete a runner."""
         try:
             if runner.instance_id:
-                self.client.terminate_instances(
-                    InstanceIds=[runner.instance_id]
-                )
-            else:
-                self.client.terminate_instances(InstanceIds=[runner.name])
+                self.client.terminate_instances(InstanceIds=[runner.instance_id])
         except ClientError as e:
-            if e.response["Error"]["Code"] == "InvalidInstanceID.NotFound":
-                log.error(f"Instance {runner.instance_id} not found.")
+            error = e.response.get("Error", {})
+            if error.get("Code") == "InvalidInstanceID.NotFound":
+                pass
+            if error.get("Code") == "InvalidInstanceID.Malformed":
+                log.error(f"Instance {runner.instance_id} malformed.")
+                pass
             else:
                 raise e
         return super().delete(runner)
 
     def get(self, instance_id: str) -> Runner:
         """Get a runner."""
+        instance = {}
         try:
             instance = self.client.describe_instances(InstanceIds=[instance_id])
-            instance_id = instance["Instances"][0]["InstanceId"]
         except ClientError as e:
-            if e.response["Error"]["Code"] == "InvalidInstanceID.NotFound":
+            error = e.response.get("Error", {})
+            if error.get("Code") == "InvalidInstanceID.NotFound":
                 log.error(f"Instance {instance_id} not found.")
```

Comment:
It's weird to continue the method after this. If we got not instance, we should return right here.

Reply:
yeah I corrected that